kore/src/Kore/Step/Axiom/Evaluate.hs

```diff
@@ -95,23 +117,31 @@ evaluateAxioms
         keepResultUnchanged = id
         introduceDefinedness = flip Pattern.andCondition
         markRemainderEvaluated = fmap TermLike.mkEvaluated
+        rules = Result.appliedRule <$> Result.results result
 
     simplifiedResults <-
-        OrPattern.simplifyPredicatesWithSmt
-           predicate (Step.gatherResults result)
+        traverse (OrPattern.simplifyPredicatesWithSmt predicate)
```

Comment:
I suggest using a `Traversal` here, so that you don't need to zip the results back up with the rules later:
```suggestion
        Lens.traverseOf (field @"results" . Lens.mapped . field @"result") 
            (OrPattern.simplifyPredicatesWithSmt predicate)
            result
```